### ISSUE TITLE
StackRox installation on AKS (1.28) via ArgoCD claims PodSecurityPolicy

### ISSUE THREAD

**Issue by mxbkmd** [stackrox/stackrox#11844]
Cluster: AKS 1.28
Stackrox 4.3.

Installation:

manual installation via Helm -> works 
It even notices:
```
 Please take note of the following:

- PodSecurityPolicies are disabled, since your environment does not support them according to API
  server properties.
```
  
 The same version (4.3), the same Helm installed via ArgoCD fails with `PodSecurityPolicy` errors:
The same error appears  for the scanner, central and central db:
  
 'The Kubernetes API could not find policy/PodSecurityPolicy for requested resource stackrox/stackrox-central. Make sure the "PodSecurityPolicy" CRD is installed on the destination cluster.'

simple `yaml` file from ArgoCD  below:

```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: stackrox-central-services
  namespace: argocd
spec:
  project: argocd-apps-project
  source:
    repoURL: 'https://charts.stackrox.io'
    targetRevision: "400.4.3"
    chart: stackrox-central-services
    helm:
      values: |
        scanner:
          disable: true
        central:
          ignoreNamespaceResources: true
          aggregationStorage:
            storageClassName: default
          persistence:
            none: true
          podSecurityPolicy:
            enabled: false
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: stackrox
  syncPolicy:
    syncOptions:
      - CreateNamespace=true
    automated:
      prune: true
      selfHeal: true 
```

section:
```

podSecurityPolicy:
            enabled: false
            
```
            
added for tests - does not help

If any additional info is needed - let me know :)

**Comment by mclasmeier** [comment]
Hello @mxbkmd,

thank you very much for your report. I investigated the situation and can share the following insights with you:

* The auto-sensing logic for PodSecurityPolicies built into our Helm chart is based on the existence of the API version `policy/v1beta1` within Helm's `.Capabilities.APIVersions` built-in object.

* I was able to verify that the content of `.Capabilities.APIVersions` actually differs, depending in whether the Helm chart ist used directly or indirectly via ArgoCD.

* More specifically, and this was somewhat surprise to me, when used indirectly via ArgoCD the API version `policy/v1beta1` is suddently contained within `.Capabilities.APIVersions`, while it is not when used Helm directly (which agrees with the output of `kubectl api-versions` when connected to a recent AKS cluster).

This explains the behavior you are seeing. I guess the ArgoCD developers can explain where this is coming from.

Besides that, there are a couple of issues in your Helm values:
```
      values: |
        scanner:
          disable: true
        central:
          ignoreNamespaceResources: true
          aggregationStorage:
            storageClassName: default
          persistence:
            none: true
          podSecurityPolicy:
            enabled: false
```

Where do these come from?
```
          ignoreNamespaceResources: true
          aggregationStorage:
            storageClassName: default
```
I don't think these configuration settings exist?

Furthermore, this part:
```
        central:
          podSecurityPolicy:
            enabled: false
```
is wrong. You can explicitly disable PSPs via:
```
        system:
          enablePodSecurityPolicies: false
```

One more thing: Unfortunately this combination (ArgoCD and AKS) will not work out of the box.
The story behind this can be read here: https://github.com/stackrox/stackrox/issues/2482.

Long story short, you need to upgrade to the most recent (published) Helm charts (4.4.x) and you need to specify this in your Helm values:

```
customize:
  other:
    persistentvolumeclaim/central-db:
      annotations:
        helm.sh/hook: null
        helm.sh/hook-delete-policy: null
        helm.sh/resource-policy: null
```

Then it should work.

Greetings
Moritz

**Comment by mclasmeier** [comment]
I will close this ticket -- please re-open in case it turns out not to be completely solved yet.
Thank you!
